components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java

```diff
@@ -216,14 +216,16 @@ public String getResponseString(OAuth2TokenValidationResponseDTO tokenResponse)
                     key = (String) keys.next();
                     String value;
                     value = newJSON.get(key).toString();
-                    JSONObject jsonObjectValues = new JSONObject(value);
-                    if (jsonObjectValues != null) {
-                        Iterator<?> claimKeyValues = jsonObjectValues.keys();
-                        while (claimKeyValues.hasNext()) {
-                            String claimKeys = (String) claimKeyValues.next();
-                            String claimValues = jsonObjectValues.get(claimKeys).toString();
-                            if (claimValues.equals("true") && claimKeys.equals("essential")) {
-                                essentailClaimslist.add(key);
+                    if (!newJSON.get(value).equals(null)) {
```

Comment:
Do you want to check whether the newJSON.get(value) is null or whether it is a string with value "null" ?

Reply:
I want to check newJSON.get(value) is null. Because the value is returned as a null value not as a string.

Reply:
Then this code will cause a null pointer exception right ? if newJSON.get(value) returns null and if u call newJSON.get(value).equals this should throw a null pointer.